### PR TITLE
Update install notes

### PR DESCRIPTION
**Pull request by jules23**
Since we last talked, I tore everything down and built everything back up on a fresh Ubuntu installation, up to having the single `test_pay_no_secret` test succeed.

This update to the notes synthesizes everything we found yesterday, and some minor details I came across today during the installation. Let me know what you think.

### CODE REVIEW

**Comment by jules23** [comment]
Note that I've removed this note here:
```
Build libsecp256k1 with `./configure --enable-module-recovery`, see
https://github.com/golemfactory/golem/issues/2168 for background.
```
As this was never something I had to deal with both yesterday and today to get the test suite running.

File: contrib/remote_hsmd/NOTES.md [hunk 1]
```diff
@@ -21,22 +11,49 @@ On Fedora:
...
+    git clone git@gitlab.com:lightning-signer/validating-lightning-signer.git
+    cargo build --features log_pretty_print,debug_enforcement_state
+
+C-Lightning:
+
+    # Add an "upstream" reference and fetch all tags (needed for pyln-{proto,client,testing})
+    # Use the HTTPS pointer if you have authentication issues with git
+    git remote add upstream git@github.com:ElementsProject/lightning.git
+    git fetch --all --recurse-submodules
+    git fetch --tags
```
Comment (ksedgwic) [inline contrib/remote_hsmd/NOTES.md:37]: There's an impending merge conflict w/ https://github.com/lightning-signer/c-lightning/pull/51, I simplified/streamlined those two lines.  Should be easy to fix after whichever PR goes first though.

File: contrib/remote_hsmd/NOTES.md [hunk 2]
```diff
@@ -21,22 +11,49 @@ On Fedora:
...
     make distclean
     ./configure --enable-developer
     make
 
-Build libsecp256k1 with `./configure --enable-module-recovery`, see
-https://github.com/golemfactory/golem/issues/2168 for background.
+On Ubuntu:
+
+    # Edit /etc/hosts, and add the entry localhost to the ::1 IP address, as below:
+    ::1     ip6-localhost ip6-loopback localhost
```
Comment (ksedgwic) [inline contrib/remote_hsmd/NOTES.md:51]: This won't be necessary after we merge https://gitlab.com/lightning-signer/validating-lightning-signer/-/merge_requests/211, but it doesn't hurt either ...

**Comment by ksedgwic** [comment]
LGTM, may need a fix if #51 merges first.

File: contrib/remote_hsmd/NOTES.md (hunk 1 above)
Comment (jules23) [reply contrib/remote_hsmd/NOTES.md:37]: Happy to rebase on top of #51 once it is merged.

File: contrib/remote_hsmd/NOTES.md (hunk 2 above)
Comment (jules23) [reply contrib/remote_hsmd/NOTES.md:51]: In that case, I'd prefer to remove these instructions once that PR is merged.

**Comment by ksedgwic** [comment]
@jules23 both PRs are merged, ready for your updates

**Pull request merged**